src/bun.js/webcore/body.zig

```diff
@@ -1258,3 +1263,345 @@ pub fn BodyMixin(comptime Type: type) type {
+        if (stream.isDone()) {
+            const bytes = sink.stream_buffer.list.items;
+            log("onStreamPipe done {}", .{bytes.len});
+            sink.onFinishedBuffering(sink.ctx, bytes, null, true);
+            return;
+        }
+    }
+
+    pub fn onResolveStream(_: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSValue {
+        var args = callframe.arguments(2);
+        var sink: *@This() = args.ptr[args.len - 1].asPromisePtr(@This());
+        sink.handleResolveStream(true);
+        return JSValue.jsUndefined();
+    }
+
+    pub fn onRejectStream(_: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSValue {
+        const args = callframe.arguments(2);
+        var sink = args.ptr[args.len - 1].asPromisePtr(@This());
+        var err = args.ptr[0];
+        sink.handleRejectStream(err, true);
+        return JSValue.jsUndefined();
+    }
+
+    fn handleRejectStream(sink: *@This(), err: JSValue, is_async: bool) void {
+        if (sink.js_sink) |wrapper| {
+            wrapper.detach();
+            sink.js_sink = null;
+            wrapper.sink.destroy();
+        }
+        sink.onFinishedBuffering(sink.ctx, "", err, is_async);
+    }
+
+    fn handleResolveStream(sink: *@This(), is_async: bool) void {
+        if (sink.js_sink) |wrapper| {
+            const bytes = wrapper.sink.bytes.slice();
+            log("handleResolveStream {}", .{bytes.len});
+            sink.onFinishedBuffering(sink.ctx, bytes, null, is_async);
+        } else {
+            log("handleResolveStream no sink", .{});
+            sink.onFinishedBuffering(sink.ctx, "", null, is_async);
+        }
+    }
+
+    fn createJSSink(sink: *@This(), stream: JSC.WebCore.ReadableStream) !void {
+        stream.value.ensureStillAlive();
+        var allocator = sink.allocator;
+        var buffer_stream = try allocator.create(ArrayBufferSink.JSSink);
+        var globalThis = sink.global;
+        buffer_stream.* = ArrayBufferSink.JSSink{
+            .sink = ArrayBufferSink{
+                .bytes = bun.ByteList.init(&.{}),
+                .allocator = allocator,
+                .next = null,
+            },
+        };
+        var signal = &buffer_stream.sink.signal;
+        sink.js_sink = buffer_stream;
+
+        signal.* = ArrayBufferSink.JSSink.SinkSignal.init(JSValue.zero);
+
+        // explicitly set it to a dead pointer
+        // we use this memory address to disable signals being sent
+        signal.clear();
+        std.debug.assert(signal.isDead());
+
+        const assignment_result: JSValue = ArrayBufferSink.JSSink.assignToStream(
+            globalThis,
+            stream.value,
+            buffer_stream,
+            @as(**anyopaque, @ptrCast(&signal.ptr)),
+        );
+
+        assignment_result.ensureStillAlive();
+
+        // assert that it was updated
+        std.debug.assert(!signal.isDead());
+
+        if (assignment_result.toError() != null) {
```

Comment:
Use `.isError()` because `toError` consumes it which potentially creates a single-use object

Reply:
fixed in https://github.com/oven-sh/bun/pull/4418/commits/7cd66913e79f7c27d13efeb30801dfcbbabe45e6